chapters/annotations.tex

```diff
@@ -1762,6 +1762,8 @@ \subsubsection{Conversion Rules}\label{conversion-rules}
 Specifying conversions using the old name of a class allows the conversion to be done without access to the old
 version of the library (by suitable modifications of the lookup).  Another alternative is to use the old version
 of the library during the conversion.
+
+An empty string would either mean an empty diagnostics, a missing class or component, or a modifier which does not make sense.
```

Comment:
It's hard to include this kind of statement about something illegal in a meaningful way.  If it should be included, I think it needs to be more clear why:
```suggestion
Not allowing the empty string just means that it is not possible to express meaningless things such as empty diagnostics, a missing class or component, or a non-sensical modifier.
```

Reply:
Although this sentence is technically correct, I do not see how it helps to clarify anything. In #3459 I was just thinking of convertElement. But it is helpful to make a full list of what cases are accepted and make it normative. 
- Any `"OldClass"`, `"oldElement"`, or `"oldName"` *must* be non-empty.  Should compilers just ignore them or result in an error?
- Should it be allowed to have `"NewClass"` as non-empty? Would it be equivalent to `convertMessage` with and empty failed message or would it be just ignored?
- Arrays for modifiers in `convertModifiers`: either one can be empty. Already explained.
- Empty `"NewName"` is the question of #3459: is equivalent to `convertModifier("OldClass","OldName",fill{"",0})`. It will result in an error if a model that is converted uses `OldClass.OldName` anywhere other than in a modification.

Reply:
Hm...
One option is to just skip this explanation for now, if the explanation isn't helpful.
Another is to spell out how to avoid using an empty string:

The following are alternatives to using an empty string:
* Instead of empty OldClass, oldElement, oldName (as a form of wild-card) list all of them.
* Instead of empty NewClass or NewElement use convertMessage to indicate why it cannot be converted.
* Instead of empty NewName for convertElement use convertModifiers("OldClass",{"OldName"},fill("",0)) - even if limited to modifications.
* Instead of empty message for convertMessage give a message explaining why.

However, looking at this list it looks more like an annotated reference than something that is needed in the specification, so I think it is easier to just drop this line.

Reply:
I believe I will just erase it.
Is that ok, @eshmoylova @henrikt-ma (and others)?

Reply:
I am ok with erasing it. But I want a sentence added to convertElement that states that case `convertElement("OldClass","OldName",fill("",0))` is equivalent to `convertModifiers("OldClass",{"OldName"},fill("",0))` and no conversion is applied to any equations.

Reply:
Ok, I have gladly removed the description of why, and reluctantly added the above.